### ISSUE TITLE
static_lifetime factories don't work reliably

### ISSUE THREAD

**Issue by danzil** [microsoft/cppwinrt#626]
make_factory() in base.h fails to find a previously registered factory with the same name. That's because it uses an hstring that has been destroyed. See line in base.h:

```
param::hstring const name{ name_of<typename D::instance_type>() };
```

The params version of hstring only keeps the handle to the string while the string returned by name_of is no longer around when the call to Lookup happens.

To reproduce this, just run App Verifier against any repro that implements a static_lifetime factory (I was on x64 if that helps).

Changing that line into this, below, fixes the problem:

```
auto const name{ name_of<typename D::instance_type>() };
```

**Comment by kennykerr** [comment]
Hi Daniel, this was an issue introduced with 2.0.200512.4 but should be fixed with 2.0.200514.2. Let me know if you still experience a problem with the latest version.

**Comment by danzil** [comment]
@kennykerr that fixed it! Thanks for the quick response as always!
